WordPress/Classes/Utility/Analytics/WPAnalyticsEvent.swift

```diff
@@ -669,4 +672,25 @@ extension WPAnalytics {
         WPAnalytics.track(stat, withProperties: props)
     }
 
+    /// This will call each registered tracker and fire the given event.
+    /// - Parameters:
+    ///   - eventName: a `String` that represents the event name
+    ///   - properties: a `Hash` that represents the properties
+    ///   - blog: a `Blog` asssociated with the event
+    static func trackBlockEditorEvent(_ eventName: String, properties: [AnyHashable: Any], blog: Blog) {
+        var props = properties
+        props[WPAppAnalyticsKeyBlogID] = blog.dotComID
+        props[WPAppAnalyticsKeySiteType] = blog.isWPForTeams() ? WPAppAnalyticsValueSiteTypeP2 : WPAppAnalyticsValueSiteTypeBlog
+
+        var event: WPAnalyticsEvent?
+        switch eventName {
+        case "editor_block_inserted": event = .gutenbergEditorBlockInserted
+        default: event = nil
+        }
+
+        if event != nil {
+             WPAnalytics.track(event!, properties: props)
+        }
```

Comment:
Could we avoid updating the props with the blog info here and instead use track(eventName, properties, blog) to take care of them?

```suggestion
    static func trackBlockEditorEvent(_ eventName: String, properties: [AnyHashable: Any], blog: Blog) {

        var event: WPAnalyticsEvent?
        switch eventName {
        case "editor_block_inserted": event = .gutenbergEditorBlockInserted
        default: event = nil
        }

        if event != nil {
             WPAnalytics.track(event!, properties: props, block: blog)
        }
```

Reply:
Definitely a good idea. Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/17243/commits/a26a8e7a6c3b8b9c516a30c579c5ba4764410fda.